### ISSUE TITLE
CSRF

### ISSUE THREAD

**Issue by kazutaka1994** [kazutaka1994/lamp_practice#8]
CSRF対策用です。

**Comment by kazutaka1994** [comment]
①function.phpに

function get_csrf_token(){
  $token = get_random_string(30);
  set_session('csrf_token', $token);
  return $token;
}

function is_valid_csrf_token($token){
  if($token === '') {
    return false;
  }
  return $token === get_session('csrf_token');
}

を追加

②

- [ ] index.php
- [ ] admin.php
- [ ] cart.php
- [ ] login.php
- [ ] signup.php

に
$token = get_csrf_token();
header('X-FRAME-OPTIONS: DENY');
を追加

③

- [ ] admin_insert.php
- [ ] admin_change_stock.php
- [ ] admin_change_status.php
- [ ] admin_delete_item.php
- [ ] cart_change_amount.php
- [ ] cart_delete_cart.php
- [ ] finish.php
- [ ] index_add_cart.php
- [ ] login_process.php
- [ ] signup_process.php

に
$token = get_post('token');
if (is_valid_csrf_token($token) === false){
  set_error('不正なアクセスです');
  redirect_to(HOME_URL);
}
を追加

**Comment by thisprodigiousgeek** [comment]
順調に設計が進んでいらっしゃいますね！
* CSRFトークンのチェックが必要なファイルの抜けがありますね！各viewに含まれているフォームのaction先を全てチェックしてみましょう。
* Issueは設計を行う場所であり、実装を行う場所ではありません。Issueにおいてはコメントと同じ感覚で、設計を日本語でまとめてみましょう！

**Comment by kazutaka1994** [comment]
コメントありがとうございます。

CSRFトークンをチェックするファイルにfinish.phpを追加します。

**Comment by thisprodigiousgeek** [comment]
はい、こちらで必要なファイルをもうらしていただけましたね！すでにコードのイメージはできていらっしゃる状態なので、実装していきましょう！
